src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1068,6 +1068,11 @@ internal int AllTypeArgumentCount()
             return count;
         }
 
+        internal ImmutableArray<TypeSymbolWithAnnotations> GetTypeParametersAsTypeArguments()
+        {
+            return this.TypeParameters.SelectAsArray(tp => TypeSymbolWithAnnotations.Create(tp.ContainingModule, tp));
```

Comment:
>TypeSymbolWithAnnotations.Create(tp.ContainingModule, tp) [](start = 59, length = 57)

Do we allow ```?``` modifier on a type parameter? Consider setting unknown nullability all the time.

Reply:
Or, perhaps not change behavior for this API. We have an invariant that type arguments should match type parameters for definitions, I am not sure how nullable state would play here, it might just create problems in some hard to predict places.

---
In reply to: [171143550](https://github.com/dotnet/roslyn/pull/25087#discussion_r171143550) [](ancestors = 171143550)

Reply:
Will address this, along with `WithoutCustomModifiers.Is() => false` as a separate PR.

---
In reply to: [171143931](https://github.com/dotnet/roslyn/pull/25087#discussion_r171143931) [](ancestors = 171143931,171143550)